### DISCUSSION TITLE
query condition on "with" relation not work

### DISCUSSION THREAD

**Discussion by sepisoltani** [jenssegers/laravel-mongodb#2265] [Q&A]
I have a Order Model which has belongsTo relation to Factor model

```php

use Jenssegers\Mongodb\Eloquent\Model;

class Order extends Model
{
  public function factor()
    {

        return $this->belongsTo(Factor::class, 'factor_id');
    }
}
```
I want to load orders which only their factor status is 'payed'.
I run this query but it loads all orders.
```php


  $res =  Order::with(['factor' => function ($query) {
            $query->where('status', '=',"payed");
        }])->get()


```

if I change the "with" , to "whereHas" it works.
why "with" doesn't work ?

**Comment by jaypalchauhan89** [comment]
With->  only relations records will be filtered.   WhereHas will filter main table's record.
